### DISCUSSION TITLE
With magick you can define the PDF version and keep it for example in version 1.3?

### DISCUSSION THREAD

**Discussion by developertrinidad08** [ImageMagick/ImageMagick#7669] [Help]
### ImageMagick version

7.1.1-38

### Operating system, version and so on

Manjaro 

### Description

Hello everyone...!!!
I'm working with Magick converting png to pdf to then send to print... the problem I found is that the printers don't support the 1.7 version of PDF (which is what it always returns to me) reading the documentation I really can't find anything about whether I can configure a version like 1.3 I would like to know if there is the possibility of defining which version of PDF I want to return when converting the png

the ideal thing for me would be to have a command like:

`magick /home/img/random.png -units PixelsPerInch -density 300 -define PDF:version=1.3 PDF:outputrandom.pdf`

But from everything I read in the documentation, a similar command does not exist or I could not find i

in the file at the following link

https://github.com/ImageMagick/ImageMagick/blob/main/coders/pdf.c#L1

I saw that it has a loop where it selects the version... but the problem with that is that no matter how much I define to fulfill what it needs to return version 1.3 it continues returning version 1.7.
making my application unstable

**Comment by snibgo** [comment]
I have no problems getting PDF v1.3:
```
D:\temp>%IMG71138%magick rose: x.pdf

D:\temp>exiftool x.pdf |grep Version
ExifTool Version Number         : 12.13
PDF Version                     : 1.3
```

**Reply by developertrinidad08** [reply]
The problem I have is that the images come from a file uploaded by the user and there are some pngs that are created as pdf v1.3 but almost all of them are created in version 1.7. More or less out of 10, 1 or 2 are created with v1.3, meaning that out of every 10 things I send to print, I print 2 (being optimistic, in reality there are fewer). 

This is why I need to define the PDF version that Magick will return when converting it.
I currently do it with Ghostscript but this adds a lot of delay before printing, it goes from taking 20 or 30 seconds to print 1 sheet to 2 or 3 minutes

**Reply by fmw42** [reply]
How is a PNG created as a PDF?

You  might find some PDF created tool that allows your to specify the version. Then perhaps create PDF write line in your delegates.xml file that calls your PDF tool with the version you want.

**Reply by developertrinidad08** [reply]
Sorry if I explained myself badly...
Basically I create a pdf with a .png image using this command:

`magick /home/img/random.png -units PixelsPerInch -density 300   PDF:outputrandom.pdf`

The truth is that I wouldn't even know where to start with the search for a PDF creation tool to help me solve this.
I thought Ghostscript would be a good solution but I didn't expect it to increase printing delay so much.

**Comment by snibgo** [comment]
> This is why I need to define the PDF version that Magick will return when converting it.

But the PDF version number that IM writes depends on features of the input image(s). For example, if the input image has transparency, then the PDF version number must be 1.4 or higher. If you want to write a PDF version 1.3, then you must first remove any transparency.

If you can find a case where the PDF version number created by IM is higher than it needs to be, then please report this.

**Reply by developertrinidad08** [reply]
Thank you very much for your help, snibgo
Once you mentioned transparency, I discovered that we can remove transparency with -alpha off
So if I first convert the image with something like this:

`magick random.png -alpha off random_truecolor.png  `

and then I convert it to PDF, so far in all my tests it returned PDF v1.3. I need to test it in more detail to see if there is a time when it doesn't. I'll let you know here although I really doubt it.
 
But for now I think this is my best option.
 
```
      cmd0 = "convert #{img} -alpha off output#{random}.png "
      cmd1 = "convert output#{random}.png -units PixelsPerInch  -density #{dpi} PDF:output#{random}.pdf" 
      cmd2 = "/usr/bin/lpr -P #{impresora} -o media=#{media} output#{random}.pdf"

```
